cdci_data_analysis/analysis/instrument.py

```diff
@@ -128,24 +129,37 @@ def set_pars_from_dic(self, par_dic, verbose=False):
             query_obj = self.get_query_by_name(query_name)
             # loop over the list of parameters for the requested query,
             # but also of the instrument query and source query
-            for par in (query_obj.parameters +
-                        self.instrumet_query.parameters +
-                        self.src_query.parameters):
+            param_list = (query_obj.parameters +
+                          self.instrumet_query.parameters +
+                          self.src_query.parameters)
+            # for par_type, group_par_type in itertools.groupby(param_list, lambda x: type(x)):
+                # for par in par_group:
+            for par in param_list:
                 # this is required because in some cases a parameter is set without a name (eg UserCatalog),
                 # or they don't have to set (eg scw_list)
-                # 
                 if par.name is not None and par.name not in params_not_to_be_included:
-                    par.set_from_form(par_dic, verbose=verbose)
-
-                self.logger.info("set_pars_from_dic>> par: %s par.name: %s par.value: %s par_dic[par.name]: %s", par, par.name, par.value, par_dic.get(par.name, None))
+                    par.set_value_from_form(par_dic, verbose=verbose)
+                self.logger.info("set_pars_from_dic>> par: %s par.name: %s par.value: %s par_dic[par.name]: %s",
+                            par, par.name, par.value, par_dic.get(par.name, None))
                 if par.name == "scw_list":
                     self.logger.info("set_pars_from_dic>> scw_list is %s", par.value)
+                # par_type.set_group_par(par_group=group_par_type,
+                #                        par_dic=par_dic,
+                #                        params_not_to_be_included=params_not_to_be_included,
+                #                        verbose=verbose)
+                # set the default, isot
+                # par_dic[par.units_name] = 'isot'
+            for par in param_list:
+                if par.name is not None and par.name not in params_not_to_be_included and par.default_units is not None:
+                    par_dic[par.units_name] = par.default_units
 
         else:
             for _query in self._queries_list:
-                for par in _query.parameters:
-                    if par.name is not None and par.name not in params_not_to_be_included:
-                        par.set_from_form(par_dic, verbose=verbose)
+                for par_type, group_par_type in itertools.groupby(_query.parameters, lambda x: type(x)):
+                    par_type.set_group_par(par_group=group_par_type,
```

Comment:
if it's removed above, is it still needed here?
by the way, this part and the one in the other branch seem partially duplicated. And it's because they are, changes need to put propagated to both manually. Would be better to de-duplicate: derive `param_list` depending on `product_type` , but loop over it only once.

Reply:
removed! forgot to remove some code